### DISCUSSION TITLE
normalization of exported bigwigs files

### DISCUSSION THREAD

**Discussion by Brawni** [GreenleafLab/ArchR#734] [Questions / Documentation] [answered]
Hello!

This is more of a naive question and not a bug but I cant specify other labels than 'bug' anymore. When exporting bigwig files using getGroupBW function wouldn't it make sense to normalise by number of cells along with the ReadsInTSS per cluster given that the coverage are represented by sum of fragments across cells ?  

Thank you!

**Comment by rcorces** [comment, marked as answer]
@Brawni - We moved questions and feature requests to the Discussions section and only bug reports go into Issues now. I'm migrating your post to Discussions.

To answer your question - I'm not sure number of cells makes sense. What if you have multiple samples in your dataset and one of them is sequenced to 1/10th the depth? Then by normalizing to number of cells you are skewing the data. The ReadsInTSS is a pretty stable normalization that ignores how many cells (they are being grouped together into a single track after all) and only cares about the total number of fragments. Thinking of it another way, if you did bulk RNA-seq on 5000 cells and on 50,000 cells, would you normalize the gene expression counts based on the number of cells input? The normal conventions (TPM/RPKM) dont take into account cell number. Same thing applies to ATAC-seq in a way though we use ReadsInTSS instead of total reads because this also accounts for data quality. That being said, you can create arbitrary columns in `cellColData` and use those for normalization to see what happens.

**Reply by Brawni** [reply]
Hi Ryan!

Thanks for your prompt reply and sorry to have posted in the wrong section. So the resulting track coverage takes the average of the sum of fragments in each cell per bin and corrects by ReadsInTSS?

**Reply by rcorces** [reply]
Not quite. `getGroupBW()` performs normalization at the pseudo-bulk level, not at the per-cell level. So it takes the sum of fragments from all cells in your group and normalizes that to the sum of ReadsInTSS from all cells in your group. The normalization is not performed on a per-cell basis.

Also, looking at the code, I dont think that arbitrary normalization based on any entry in `cellColData` is allowed (I misspoke before). It can only be `ReadsInTSS` `ReadsInPromoters`  `nFrags` or `None`:

https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/GroupExport.R#L352

**Reply by Brawni** [reply]
Got it! Thanks!!
